### PR TITLE
Add support for start_date, deprecate start_year

### PR DESCRIPTION
**Pull request by berland**
Solves #40

### CODE REVIEW

File: stea/stea_input.py [hunk 1]
```diff
@@ -67,7 +67,9 @@ def __init__(self, argv):
             self.config = config
 
         except Exception as ex:
-            raise ValueError(f"Could not load config file: {args.config_file}") from ex
+            raise ValueError(
+                f"Could not load config file: {args.config_file}, error: {ex}"
```
Comment (berland) [inline stea/stea_input.py:71]: This is not a pretty change, but allows `pytest.raises(match="something")` to work on the inner exception.

**Comment by kvashchuka** [comment]
@berland I believe README should also be updated. I am thinking about this line:
```
but you can optionally use the keywords start-year and end-year to limit the time range.
```

**Comment by berland** [comment]
> @berland I believe README should also be updated. I am thinking about this line:
> 
> ```
> but you can optionally use the keywords start-year and end-year to limit the time range.
> ```

done

**Comment by kvashchuka** [comment]
It is outside of the scope of this PR, but why we don't make the same change for the `end_year`? Is it not a problem or is it just coming later?

File: stea/stea_keys.py [hunk 2]
```diff
@@ -44,7 +44,8 @@ class SteaInputKeys:
     ECL_MULT = "mult"
     ECL_GLOB_MULT = "glob_mult"
     SERVER = "stea_server"
-    START_YEAR = "start_year"
+    START_DATE = "start_date"
+    START_YEAR = "start_year"  # Deprecated
```
Comment (kvashchuka) [inline stea/stea_keys.py:48]: I am a bit confused - is the plan for `START_YEAR` to stay for now with deprecation warning and we delete it later?

Comment (berland) [reply stea/stea_keys.py:48]: Yes.

**Comment by berland** [comment]
> It is outside of the scope of this PR, but why we don't make the same change for the `end_year`? Is it not a problem or is it just coming later?

[discussed orally] For reference: `end_year` was not requested, and in terms of deprecation calculation, the start and end of the time range are not symmetrical (start is far more important than the end).

**Review by kvashchuka — COMMENTED** [review]
Looks good! Thank you for your patience with me :) there is just one comment left

File: stea/stea_request.py [hunk 3]
```diff
@@ -69,17 +70,26 @@ def add_ecl_profile(
...
+        if start_date is None:
             start_date = case.start_date
-            first_year = start_date.year
 
-        if last_year is None:
+        if end_year is None:
             end_date = case.end_date
-            last_year = end_date.year
+        else:
+            end_date = datetime.date(end_year, 1, 1)
```
[the remark this answers is not in the document]
Comment (berland) [reply stea/stea_request.py:79]: Good point. I am sure it is actually ignored by `case.time_range()`, but I will fix it and document it!

[the remark this answers is not in the document]
Comment (berland) [reply stea/stea_request.py]: done.

**Review by kvashchuka — APPROVED** [review]
LGTM

**Pull request merged**